packages/design-system/src/scalars/components/number-field/number-field.tsx

```diff
@@ -67,6 +70,42 @@ export const NumberFieldRaw = forwardRef<HTMLInputElement, NumberFieldProps>(
       precision,
     );
 
+    const handleChange = (
+      e: React.MouseEvent<HTMLButtonElement>,
+      operation: "increment" | "decrement",
+    ) => {
+      e.preventDefault();
+      const currentValue = Number(displayValue ?? defaultValue ?? 0);
+
+      const adjustment = (operation === "increment" ? 1 : -1) * (step || 1);
+      const newValue = currentValue + adjustment;
+
+      if (
+        (maxValue !== undefined && newValue > maxValue) ||
+        (minValue !== undefined && newValue < minValue)
+      ) {
+        return;
+      }
+      if (newValue === currentValue) {
+        return;
+      }
+      const nativeEvent = new Event("input", {
```

Comment:
`input` and `change` events are different. In  this case we're trying to trigger a "change" event instead

Reply:
you are right, done